mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -922,10 +922,12 @@ public bool ConnectAsync (SocketAsyncEventArgs e)
 				}
 
 				if (ares.IsCompleted && ares.CompletedSynchronously) {
+					Interlocked.Exchange (ref e.in_progress, 0);
```

Comment:
Does it make sense to use Interlocked here?

Reply:
I will change it.